core/shared/src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -181,6 +182,23 @@ object Argument extends PlatformArguments {
       } catch { case _: IllegalArgumentException => Validated.invalidNel(s"Invalid UUID: $string") }
     }
 
+  implicit val readDuration: Argument[Duration] =
+    from("duration") { string =>
+      try Validated.Valid(Duration(string))
+      catch { case _: NumberFormatException => Validated.invalidNel(s"Invalid Duration: $string") }
+    }
+
+  implicit val readFiniteDuration: Argument[FiniteDuration] =
+    new Argument[FiniteDuration] {
+      override def read(string: String): ValidatedNel[String, FiniteDuration] =
+        readDuration.read(string) andThen {
```

Comment:
Here we reuse the normal `Duration` reader however we make sure that its only finite using the `andThen`